pkg/ottl/contexts/internal/span.go

```diff
@@ -300,6 +309,64 @@ func accessKind[K SpanContext]() ottl.StandardGetSetter[K] {
 	}
 }
 
+func accessStringKind[K SpanContext]() ottl.StandardGetSetter[K] {
+	return ottl.StandardGetSetter[K]{
+		Getter: func(ctx context.Context, tCtx K) (interface{}, error) {
+			return tCtx.GetSpan().Kind().String(), nil
+		},
+		Setter: func(ctx context.Context, tCtx K, val interface{}) error {
+			if s, ok := val.(string); ok {
+				var kind ptrace.SpanKind
+				switch s {
+				case "Internal":
+					kind = ptrace.SpanKindInternal
+				case "Server":
+					kind = ptrace.SpanKindServer
+				case "Client":
+					kind = ptrace.SpanKindClient
+				case "Producer":
+					kind = ptrace.SpanKindProducer
+				case "Consumer":
+					kind = ptrace.SpanKindConsumer
+				default:
+					kind = ptrace.SpanKindUnspecified
```

Comment:
Do we have any validation on this set of values? Or will it silently set `SpanKindUnspecified ` if a user has `span_kind.string == "consumer"`  in their config?

Reply:
It would set `SpanKindUnspecified`.  I'll add a case for `SpanKindUnspecified` specifically and update the default case to print before setting `SpanKindUnspecified`.  But I don't want to do any sort of case-insensitive comparing and guessing of what the user was trying to set.

Reply:
@dmitryax added a print statement (we don't have access to the logger here).  Here is what it looks like when hit with a telemetrygen trace with 2 spans:

```yaml
processors:
  transform:
    error_mode: ignore
    trace_statements:
      - context: span
        statements:
          - set(kind.string, "foo")
          - set(kind.deprecated_string, "foo")
```

```
2023-05-16T09:30:11.773-0600    info    service/service.go:158  Everything is ready. Begin running and processing data.
unknown span kind, foo, setting Unspecified instead.
unknown span kind, foo, setting Unspecified instead.
unknown span kind, foo, setting Unspecified instead.
unknown span kind, foo, setting Unspecified instead.
2023-05-16T09:30:14.840-0600    info    TracesExporter  {"kind": "exporter", "data_type": "traces", "name": "logging", "resource spans": 1, "spans": 2}
```

Reply:
I don't think we should apply fmt.Print. Is there a way to pass logger to the pkg/ottl? It's not for this PR, but I think it's something that we need to think about for future. I expect  more cases when the logger would be required.

Another Q: why do we need to set undefined on the incorrect input? As a user, I would expect this operation to be ignored with an error log message, ideally during configuration validation if feasible.

Reply:
I'm glad you're asking these questions and keeping me on point.

I agree the contexts need a logger.  I looked at this the other day and it wouldn't be trivial, so definitely a future PR.

Looking through the other accessors I did deviate from the norm on this one so thanks for reeling me back in. All the accessors in OTTL currently set nothing and do not error if they get the incorrect type.  If they do get the right type and they are unable to set the value then they set nothing and return the error; setting `span_id` and `trace_id` are an example of this.

With that in mind, I agree that the most consistent behavior would be to use the switch statement to check for expected values and if we get an unexpected value to set nothing and return an error.

Reply:
Sounds good. Can you please create an issue for passing the logger?

Reply:
Done https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/22076